azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/state/VisibilityState.kt

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.redux.state
+
+internal enum class VisibilityStatus {
+    VISIBLE,
+    HIDE_REQUESTED,
```

Comment:
This might be asking to much in a PR, however.

This would feel much better if it was 
States = [VISIBLE, GONE, PIP]


and didn't have those intermediary states like HIDE_REQUESTED.

If the intermediary is because we need code like oldState.visibilityStatus != newState.visibilityStatus, it's probably worth talking middleware again.

Reply:
HIDE_REQUESTED is a command that goes through middleware, reducer to state as a communication between components. It is needed here.